components/proxy/src/main/java/com/hotels/styx/StyxConfig.java

```diff
@@ -64,10 +66,9 @@ public StyxConfig(String yaml) {
     public StyxConfig(StartupConfig startupConfig, Configuration configuration) {
         this.startupConfig = requireNonNull(startupConfig);
         this.configuration = requireNonNull(configuration);
-        this.adminServerConfig = get("admin", AdminServerConfig.class).orElseGet(AdminServerConfig::new);
-        this.proxyServerConfig = get("proxy", ProxyServerConfig.class).orElseGet(ProxyServerConfig::new);
-
-        this.styxHeaderConfig = get("styxHeaders", StyxHeaderConfig.class).orElseGet(StyxHeaderConfig::new);
+        this.proxyServerConfig = lazy(() -> get("proxy", ProxyServerConfig.class).orElseGet(ProxyServerConfig::new));
+        this.adminServerConfig = lazy(() -> get("admin", AdminServerConfig.class).orElseGet(AdminServerConfig::new));
+        this.styxHeaderConfig = lazy(() -> get("styxHeaders", StyxHeaderConfig.class).orElseGet(StyxHeaderConfig::new));
```

Comment:
This is not an optimisation, the code doesn't work without this.
This is because validation is applied to the `StyxConfig` object after construction.
We should not try to deserialise objects like `proxy`, `admin`, etc. before validation (in case of invalid config, that results in the Exceptions we want to avoid).

Reply:
Okay. I will have another look.